### PR TITLE
Task/jm 5900 - Pool Status Report

### PR DESCRIPTION
**Pull request by philhead1**
### JIRA link (if applicable) ###
https://centralgovernmentcgi.atlassian.net/browse/JM-5900

### Change description ###
Added new report type "PoolStatusReport"

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/juror/api/moj/report/DataType.java [hunk 1]
```diff
@@ -76,8 +78,49 @@ public enum DataType implements IDataType {
         QBulkPrintData.bulkPrintData),
     TOTAL_SENT_FOR_PRINTING("Sent for printing", Long.class, QBulkPrintData.bulkPrintData.jurorNo.count(),
         QBulkPrintData.bulkPrintData),
-    DATE_SENT("Date sent", LocalDate.class, QBulkPrintData.bulkPrintData.creationDate,  QBulkPrintData.bulkPrintData);
+    DATE_SENT("Date sent", LocalDate.class, QBulkPrintData.bulkPrintData.creationDate, QBulkPrintData.bulkPrintData),
 
+    IS_ACTIVE("Is active", Boolean.class,
```
[the remark this answers is not in the document]
Comment (philhead1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/report/DataType.java:83]: There was no usages for this in the end, removed

File: src/main/java/uk/gov/hmcts/juror/api/moj/report/DataType.java [hunk 2]
```diff
@@ -76,8 +78,49 @@ public enum DataType implements IDataType {
...
+        new CaseBuilder().when(QJurorStatus.jurorStatus.status.loe(4))
+            .then(true)
+            .otherwise(false),
+        QJurorStatus.jurorStatus),
+
+    SUMMONS_TOTAL("Summoned", Integer.class,
+        new CaseBuilder().when(QJurorPool.jurorPool.status.status.eq(IJurorStatus.SUMMONED)).then(1).otherwise(0).sum(),
```
[the remark this answers is not in the document]
Comment (philhead1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/report/DataType.java:90]: There's no existing join for the QJurorStatus, it's possible but arguably not needed as the entity provides the join.

File: src/main/java/uk/gov/hmcts/juror/api/moj/report/standard/PoolStatusReport.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    @Override
+    public Map<String, AbstractReportResponse.DataTypeValue> getHeadings(StandardReportRequest request,
+                                     AbstractReportResponse.TableData<List<LinkedHashMap<String, Object>>> tableData) {
+        Optional<PoolRequest> optionalPoolRequest =
+            getPoolRequestRepository().findByPoolNumber(request.getPoolNumber());
+        if (optionalPoolRequest.isEmpty()) {
+            throw new MojException.NotFound("Cannot find pool number " + request.getPoolNumber(), null);
+        }
```
[the remark this answers is not in the document]
Comment (philhead1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/report/standard/PoolStatusReport.java:65]: Updated 😄

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/reports/request/StandardReportRequest.java [hunk 4]
```diff
@@ -40,8 +40,9 @@ public class StandardReportRequest {
         "PanelListDetailedReport",
         "AbaccusReport",
         "JuryListReport",
+        "PoolStatusReport",
         //Grouped
-        "PostponedListByDateReport"
+        "PostponedListByDateReport",
```
[the remark this answers is not in the document]
Comment (philhead1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/controller/reports/request/StandardReportRequest.java]: Removed 😄

**Review by CCHAPMAN01 — APPROVED** [review]

**Pull request merged**